shared/AppInsightsCommon/src/StorageHelperFuncs.ts

```diff
@@ -23,6 +24,53 @@ function _getLocalStorageObject(): Storage {
     return null;
 }
 
+/**
+ * Safely checks if storage object (localStorage or sessionStorage) is available and accessible
+ * This helps prevent SecurityError in some browsers (e.g., Safari) when cookies are blocked
+ * @param storageType - Type of storage
+ * @returns {boolean} Returns whether storage object is safely accessible
+ */
+function _canSafelyAccessStorage(storageType: StorageType): boolean {
+    const storageTypeName = storageType === StorageType.LocalStorage ? "localStorage" : "sessionStorage";
+    
+    try {
+        // First, check if window exists and get the global object once
+        const gbl: any = getGlobal();
+        if (isNullOrUndefined(gbl)) {
+            return false;
```

Comment:
refactor the code to only have a single return at the end, using local "result" variable which defaults to true